GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java

```diff
@@ -445,13 +448,22 @@ public void bindShaders(GVRSceneObject root) {
      * @see GVRScene.getLightList
      */
     protected boolean addLight(GVRLightBase light) {
-        if (light != null) {
-            Integer lightIndex = mLightList.size();
-            String name = "light" + lightIndex.toString();
-            mLightList.add(light);
-            NativeLight.setLightID(light.getNative(), name);
-            NativeScene.addLight(getNative(), light.getNative());
-            return true;
+        if (light == null)
+        {
+        	return false;
```

Comment:
In case of null I'd say either let it throw the NPE later or throw an IllegalArgumentException from here. Checking for null seems better left to the caller.